### PR TITLE
Add Bluetooth Device Information Service with PNP_ID

### PR DESCRIPTION
**Pull request by angusjfw**
| Status  | Ticket/Issue |
| :---: | :--: |
| Ready/Hold | [Ticket](https://pi-top.atlassian.net/browse/<ticket-number>) |

#### Main changes

- Add standard DIS UUIDs according to Bluetooth SIG spec
- Implement the Device Information Service (DIS) with manufacturer name, model number, and PNP_ID
- Include device info and versioning in PNP_ID structure
- Add the DIS service to the Bluetooth server

🤖 Generated with [Claude Code](https://claude.ai/code)

#### Screenshots (feature, test output, profiling, dev tools etc)

[insert screenshots here]

#### Other notes (e.g. implementation quirks, edge cases, questions / issues)
-

#### Manual testing tips
-

#### Tag anyone who definitely needs to review or help
-

### CODE REVIEW

File: debian/further-link.postinst [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+###############################################################
+
+case "${1}" in
+  configure)
+
+    CONFIG_FILE="/etc/bluetooth/main.conf"
+    if [ -f "$CONFIG_FILE" ]; then
+        # First, check if DeviceID is already set (to avoid duplication)
+        if grep -q '^\s*DeviceID\s*=' "$CONFIG_FILE"; then
+            echo "DeviceID already set in $CONFIG_FILE. No changes made."
```
Comment (angusjfw) [inline debian/further-link.postinst:17]: if DeviceID is set to something other than false we should probably change it

File: further_link/util/bluetooth/dis_service.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+    "1",
+    "true",
+) or state.get("bluetooth", "encrypt", fallback="0").lower() in ("0", "false"):
+    logging.info("Using unencrypted bluetooth characteristics for DIS")
+    CharFlags = NonSecureFlags
+
+
+MANUFACTURER = "pi-top"
+MODEL_NUMBER = "pi-top [4]"
+VENDOR_ID = 0x0590
```
Comment (angusjfw) [inline further_link/util/bluetooth/dis_service.py:40]: this is Nordic, let's use 0x0A5C for Broadcom

File: further_link/util/bluetooth/dis_service.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+PRODUCT_ID = 0x0001
```
Comment (angusjfw) [inline further_link/util/bluetooth/dis_service.py:41]: 0x434C for rpi 4

File: further_link/util/bluetooth/pairing.py [hunk 4]
```diff
@@ -69,4 +120,5 @@ def main():
 
 
 if __name__ == "__main__":
+    logging.basicConfig(level=logging.DEBUG)
```
Comment (angusjfw) [inline further_link/util/bluetooth/pairing.py:123]: meant to be removed?

File: further_link/util/bluetooth/server.py [hunk 5]
```diff
@@ -1,56 +1,101 @@
...
+            self.services.append(service_instance)
+
     def get_service(self, uuid: str) -> Optional[Service]:
-        for service in self.services:
-            if service.UUID == uuid:
-                return service
-        return None
+        return find_object_with_uuid(self.services, uuid)
+
+    async def _configure_gap(self):
```
Comment (angusjfw) [inline further_link/util/bluetooth/server.py:48]: we don't think this is necessary and it's odd that server name and appearance are duplicated with the advertisement.

do we keep it?

File: further_link/util/bluetooth/server.py [hunk 6]
```diff
@@ -1,56 +1,101 @@
...
         logging.debug("Starting bluetooth server...")
         self.bus = await get_message_bus()
-        adapter = await Adapter.get_first(self.bus)
+        self._adapter = await Adapter.get_first(self.bus)
+
+        # Configure GAP properties
+        await self._configure_gap()
 
-        # DEBUG: avoids use of miniscreen for pairing
+        # Configure advertisement
```
Comment (angusjfw) [inline further_link/util/bluetooth/server.py:75]: explain this is mainly for development and reuse the pairing class

File: further_link/util/bluetooth/values.py [hunk 7]
```diff
@@ -0,0 +1,9 @@
+MANUFACTURER = "pi-top"
+MODEL_NUMBER = "pi-top [4]"
+VENDOR_ID = 0x0590
```
Comment (angusjfw) [inline further_link/util/bluetooth/values.py:3]:
```suggestion
VENDOR_ID =  0x0A5C
```

File: further_link/util/bluetooth/server.py [hunk 8]
```diff
@@ -1,56 +1,83 @@
...
+        self._adapter = await Adapter.get_first(self.bus)
 
-        # DEBUG: avoids use of miniscreen for pairing
-        if environ.get("FURTHER_LINK_BLUETOOTH_PAIR_AND_ADVERTISE") in ("1", "true"):
+        # For development purposes; keep pairing mode on if env var is set
+        if environ.get("FURTHER_LINK_BLUETOOTH_PAIR_AND_ADVERTISE", "0").lower() in (
+            "1",
+            "true",
+        ):
             advert = Advertisement(
```
Comment (angusjfw) [inline further_link/util/bluetooth/server.py:59]: should use the new pairing class

**Review by angusjfw — COMMENTED** [review]
should reuse pairing class for dev pairing mode, otherwise great
